app/src/components/domain/RiskImminentEvents/Gdacs/index.tsx

```diff
@@ -83,6 +130,50 @@ function Gdacs(props: Props) {
         apiType: 'risk',
         url: '/api/v1/gdacs/{id}/exposure/',
         pathVariables: ({ eventId }) => ({ id: Number(eventId) }),
+        onSuccess: (res) => {
+            if (isNotDefined(res)) {
+                return defaultLayersValue;
+            }
+
+            const { footprint_geojson } = res;
+
+            if (isNotDefined(footprint_geojson)) {
+                return defaultLayersValue;
+            }
+
+            const footprint = isValidFeatureCollection(footprint_geojson)
+                ? footprint_geojson
+                : undefined;
+
+            if (!footprint) {
+                return defaultLayersValue;
+            }
+            const updatedLayers = {} as typeof defaultLayersValue;
+            footprint.features.reduce((_, feature) => {
```

Comment:
We could use `forEach` here. Also, the way of determining if the layers exist or not is quite error prone. Let's think about better approaches if there are any.